forge/test/mlir/mnist/training/test_training.py

```diff
@@ -469,3 +469,79 @@ def test_lora():
         test_loss += tt_loss(pred, target)[0]
 
     print(f"Test (total) loss: {test_loss}")
+
+
+@pytest.mark.push
+def test_e2e_device():
```

Comment:
Just add '_gradient_accumulation' to the test name (or push optimizer.step() inside the batch loop).